### PR TITLE
First draft MSI as a FIC sample

### PR DESCRIPTION
**Pull request by dkershaw10**
Subsequent update will finish the README.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoftgraph/msgraph-bicep-types/pull/201)

### CODE REVIEW

File: quickstart-templates/msi-as-a-fic-secretless/main.bicep [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+// The script optionally
+
+@description('Specifies the name of cloud environment to run this deployment in.')
+@allowed(['publicCloud','usGov','china','usNat','usSec'])
+param cloudEnvironment string
+
+// NOTE: Microsoft Graph Bicep file deployment is only supported in Public Cloud
+@description('Audience uris for public and national clouds')
+param audiences object = {
+  publicCloud: {
```
Comment (jason-dou) [inline quickstart-templates/msi-as-a-fic-secretless/main.bicep:15]: Bicep has an `environment()` function and its output contains the name of current cloud https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/bicep-functions-deployment#example-1. Should we use that as the key in this object?

Comment (dkershaw10) [reply quickstart-templates/msi-as-a-fic-secretless/main.bicep:15]: Will investigate -  I use it elsewhere - not sure why it didn't occur to me.

Comment (dkershaw10) [reply quickstart-templates/msi-as-a-fic-secretless/main.bicep:15]: Difficult to know what the names are for USSec and USNat. Those two are not documented from what I can tell, and using Az CLI/PS only public, china and USGov cloud endpoints show up in azure environment list response.

Would only be able to test this in USSec and USNat in those tenants in that environment.  Will ask some internal folks...

Comment (dkershaw10) [reply quickstart-templates/msi-as-a-fic-secretless/main.bicep:15]: @jason-dou best guess based on documentation on Azure Government Clouds, but I don't know the "official" Azure names for USNat and USSec. USNat looks like it's USGov, but with 2 regions exclusively for DoD (which I assume are air-gapped). USSec looks altogether different, but then there's also top secret!  Very confused.  See https://azure.microsoft.com/en-us/blog/azure-government-top-secret-now-generally-available-for-us-national-security-missions/

Comment (dkershaw10) [reply quickstart-templates/msi-as-a-fic-secretless/main.bicep:15]: Got confirmation from our contacts on the naming.  Will update

Comment (dkershaw10) [reply quickstart-templates/msi-as-a-fic-secretless/main.bicep]: Updated after confirmation from the AGC and ARM teams who corroborated the values.

File: quickstart-templates/msi-as-a-fic-secretless/secretless-graph-request.ps1 [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#######################################################################
+<#
+Perform a 2-legged flow to acquire an access token with a managed
+identity as the credential and then use it to call Microsoft Graph.
```
Comment (jason-dou) [inline quickstart-templates/msi-as-a-fic-secretless/secretless-graph-request.ps1:4]: nit: maybe note that this script is for public cloud, and the `resourceUrl` and Graph endpoint needs to be configured if need to run for other clouds

**Review by jason-dou — APPROVED** [review]

**Review by jason-dou — DISMISSED** [review]

**Review by jason-dou — APPROVED** [review]

**Pull request merged**